buildbot/osuosl/master/master.cfg

```diff
@@ -94,25 +101,29 @@ c['services'] = config.status.getReporters()
 
 ####### PROJECT IDENTITY
 
-c['title'] = config.options.get('Buildbot', 'title')
-c['titleURL'] = config.options.get('Buildbot', 'title_url')
-c['buildbotURL'] = config.options.get('Buildbot', 'my_url')
+c['title'] = "LLVM Buildbot (local testing mode)" if test_mode else config.options.get('Buildbot', 'title')
+c['titleURL'] = "http://localhost:8011" if test_mode else config.options.get('Buildbot', 'title_url')
+c['buildbotURL'] = "http://localhost:8011" if test_mode else config.options.get('Buildbot', 'my_url')
 
 # minimalistic config to activate new web UI
-c['www'] = dict(port=8011,
-                plugins=dict(waterfall_view={}, console_view={}, grid_view={}), # TODO: badges
-                default_page='console',
-                auth=config.auth.getAuth(),
-                authz=config.auth.getAuthz(),
-                #logRotateLength=
-                #maxRotatedFiles=
-                #versions=
+www_config = dict(port="tcp:8011:interface=127.0.0.1" if test_mode else 8011,
+                  plugins=dict(waterfall_view={}, console_view={}, grid_view={}), # TODO: badges
+                  default_page='console',
+                  #logRotateLength=
+                  #maxRotatedFiles=
+                  #versions=
             )
 
+if not test_mode:
```

Comment:
If we dont get auth from github, does it mean that all worker are accessible to us or are there other changes required for allowing test buildmaster owner to be the owner of all the connected workers.

Reply:
Leaving out both auth and authz fields in the config dict seems to result in being able to access the web interface with no auth needed, and anyone accessing the interface being able to force builds and clear queues. Leaving out 'auth' but leaving in 'authz' means you can't do those commands.

Are there any other actions I might have missed I should test?

The `buildbotURL` also needs to be correct or else you get "invalid origin" errors when clicking those buttons in the web UI. This is another case ssh port forwarding helps if you happened to be testing on a server - as you can have it so `localhost:8011` forwards to 8011 on the loopback interface on the remote machine.